src/material-experimental/dialog/dialog-injectors.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {InjectionToken} from '@angular/core';
+import {ComponentType, Overlay, ScrollStrategy, BlockScrollStrategy} from '@angular/cdk/overlay';
+import {DialogRef} from './dialog-ref';
+import {CdkDialogContainer} from './dialog-container';
+import {DialogConfig} from './dialog-config';
+
+/** Injection token for the Dialog's ScrollStrategy. */
+export const DIALOG_SCROLL_STRATEGY =
+    new InjectionToken<() => ScrollStrategy>('DialogScrollStrategy');
+
+/** Injection token for the Dialog's Data. */
+export const DIALOG_DATA = new InjectionToken<any>('DialogData');
+
+/** Injection token for the DialogRef constructor. */
+export const DIALOG_REF = new InjectionToken<DialogRef<CdkDialogContainer>>('DialogRef');
```

Comment:
The generic param in `DialogRef` is the component that we instantiated for the consumer, not `CdkDialogContainer`.